src/components/component-playground.js

```diff
@@ -156,26 +157,42 @@ class ComponentPlayground extends Component {
     };
   }
 
+  static getDerivedStateFromProps(nextProps, preState) {
+    const updatedState = {};
+    if (nextProps.code !== preState.code) {
+      const code = (nextProps.code || defaultCode).trim();
+      updatedState.code = code;
+    }
+    if (nextProps.scope !== preState.scope) {
+      const scope = getEnhancedScope(nextProps.scope);
+      updatedState.scope = scope;
+    }
+    return isEmpty(updatedState) ? null : updatedState;
+  }
+
   componentDidMount() {
     localStorage.setItem(STORAGE_KEY, this.state.code);
     window.addEventListener('storage', this.syncCode);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.code !== this.props.code) {
-      const code = (this.props.code || defaultCode).trim();
-      this.setState({ code });
-    }
-    if (nextProps.scope !== this.props.scope) {
-      const scope = getEnhancedScope(nextProps.scope);
-      this.setState({ scope });
-    }
+  componentDidUpdate() {
+    this.playgroundsetState();
   }
 
   componentWillUnmount() {
     window.removeEventListener('storage', this.syncCode);
   }
 
+  playgroundsetState() {
```

Comment:
Small nit: could you update this to proper camelCase? I think `playgroundSetState` is just a bit more pronounced